source/LimitedEvents.h

```diff
@@ -0,0 +1,110 @@
+/* LimitedEvents.h
+Copyright (c) 2023 by an anonymous author.
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIMITEDEVENTS_H_
+#define LIMITEDEVENTS_H_
+
+#include "Random.h"
+#include "RandomEvent.h"
+
+#include <string>
+
+
+// A class that represents an event that triggers randomly, but only a
+// limited number of these events can be active at once. This is
+// identified by the id, which is randomly generated if one is not
+// provided.  When the event first becomes possible (like entering a
+// system) then the event should immediately trigger initialCount
+// number of times.
+template <typename T>
+class LimitedEvents: public RandomEvent<T> {
+public:
+	static const int NO_LIMIT = -1;
```

Comment:
I see this variable used only once to default another variable, makes it unecessary in my opinion, but maybe I am overlooking something?

Reply:
The caller would need to know that a negative limit means "no limit," rather than having a static const int to hide the implementation. However, I think your suggestion is okay since the only numbers left are negative ones (0=do not spawn, \>0=spawn this many).

Reply:
Actually, while rearranging things to follow your other suggestions, I now use NO_LIMIT in several places (will push soon), so I'll need to keep it.